apps/cli/src/commands/doctor.ts

```diff
@@ -44,18 +44,39 @@ export default class DoctorCommand extends Command {
             }
 
             // Check Docker Compose version
-            const { stdout: dockerComposeVersionOutput } = await execa(
-                "docker",
-                ["compose", "version", "--short"],
-            );
-            const dockerComposeVersion = dockerComposeVersionOutput.trim();
-
-            if (
-                !DoctorCommand.isDockerComposeVersionValid(dockerComposeVersion)
-            ) {
-                throw new Error(
-                    `Unsupported Docker Compose version. Minimum required version is ${DoctorCommand.MINIMUM_DOCKER_COMPOSE_VERSION}. Installed version is ${dockerComposeVersion}.`,
+            try {
+                // Attempt to get the version of Docker Compose
+                const { stdout: dockerComposeVersionOutput } = await execa(
+                    "docker",
+                    ["compose", "version", "--short"],
                 );
+
+                const dockerComposeVersion = dockerComposeVersionOutput.trim();
+
+                // Check if the Docker Compose version is valid
+                if (
+                    !DoctorCommand.isDockerComposeVersionValid(
+                        dockerComposeVersion,
+                    )
+                ) {
+                    throw new Error(
+                        `Unsupported Docker Compose version. Minimum required version is ${DoctorCommand.MINIMUM_DOCKER_COMPOSE_VERSION}. Installed version is ${dockerComposeVersion}.`,
+                    );
+                }
+
+                // If execution reaches this point, Docker Compose is installed and its version is valid
+            } catch (error: any) {
+                // Specify the type of error as `any` or `unknown` or `Error`
+                // Check if the error indicates that Docker Compose is not recognized as a command
+                if (error.message.includes("Command failed")) {
```

Comment:
I think this is too error prone. In the `try` block you run other commands.

Fox ex., if I try to run `sunodo doctor` without `docker` installed, I get a weird error message.

```shell
❯ sunodo-dev doctor
 ›   Error: Command failed with ENOENT: docker version --format {{json .Client.Version}}
 ›   spawn docker ENOENT
```

I think it's better to isolate in different blocks.

- Check if docker in installed
- Check if docker has the >= required version
- Check is compose is isntalled
- Check is compose is the >= required version

Reply:
And maybe using the error code instead of relying on the error message content.